libs/state/selections/src/lib/operators/select.ts

```diff
@@ -222,7 +223,7 @@ export function select<T extends Record<string, unknown>>(
       return state$.pipe(
         selectSlice<T & object, keyof T>(
           opOrMapFn[0] as (keyof T)[],
-          opOrMapFn[2]
+          opOrMapFn[2] as KeyCompareMap<{ [P in keyof T]: (T & object)[P] }>
         ),
```

Comment:
just a thought, wouldn't it be possible to insert that logic into this branch? we only need to omit calling the `map` function when it's not given

Reply:
@hoebbelsB sure good idea